app/logic/emailHandler2.py

```diff
@@ -0,0 +1,189 @@
+from flask import request
+from urllib.parse import urlparse
+from datetime import datetime
+from flask_mail import Mail, Message
+
+from app import app
+from app.models.programEvent import ProgramEvent
+from app.models.interest import Interest
+from app.models.user import User
+from app.models.program import Program
+from app.models.eventParticipant import EventParticipant
+from app.models.emailTemplate import EmailTemplate
+from app.models.emailLog import EmailLog
+from app.models.event import Event
+
+class EmailHandler:
+    def __init__(self, raw_form_data):
+        self.mail = Mail(app)
+        self.raw_form_data = raw_form_data
+        self.override_all_mail = app.config['MAIL_OVERRIDE_ALL']
+        self.template_identifier = None
+        self.subject = None
+        self.body = None
+        self.reply_to = None
+        self.event = None
+        self.program_ids = None
+        self.recipients = None
+        self.sl_course_id = None
+
+
+    def process_data(self):
+        """ Processes raw data and stores it in class variables to be used by other methods """
+        # Email Template Data
+        self.template_identifier = self.raw_form_data['templateIdentifier']
+
+        if 'subject' in self.raw_form_data:
+            self.subject = self.raw_form_data['subject']
+
+        if 'body' in self.raw_form_data:
+            self.body = self.raw_form_data['body']
+
+        if 'replyTo' in self.raw_form_data:
+            self.reply_to = self.raw_form_data['replyTo']
+
+        # Event
+        if 'eventID' in self.raw_form_data:
+            event = Event.get_by_id(self.raw_form_data['eventID'])
+            self.event = event
+
+        # Program
+        if 'programID' in self.raw_form_data:
+            self.program_ids = self.fetch_event_programs(self.raw_form_data['programID'])
+
+        # Recipients
+        if 'recipientsCategory' in self.raw_form_data:
+            self.recipients_category = self.raw_form_data['recipientsCategory']
+            self.recipients = self.retrieve_recipients(self.recipients_category)
+
+        # Service Learning Course
+        if 'slCourseId' in self.raw_form_data:
+            self.sl_course_id = self.raw_form_data['slCourseId']
+
+    def fetch_event_programs(self, programId):
+        """ Fetches all the programs of a particular event """
+        # Non-student-led programs have "Unknown" as their id
+        # ---Q: maybe this id should be changed to something more specific?
+        if programId == 'Unknown':
+            programs = ProgramEvent.select(ProgramEvent.program).where(ProgramEvent.event==self.event.id)
+            return [program.program for program in programs.objects()]

```

Comment:
I would rename the variable in 68 as `programEvents`, since that is the object it is fetching.